File: algorithms/leet.2180.src.1.py
class Solution:
    def countEven(self, num: int) -> int:
        ans = 0
        for x in range(2, num+1):
            yes = 1
            for c in str(x):
                if c in '13579':
                    yes ^= 1
            ans += yes
        return ans
                    

File: algorithms/leet.2180.src.1.go
func countEven(num int) int {
    // 从 0 出发，肯定成对出现，因此从 num-1
    cnt := 0
    if num % 2 == 0 {
        cnt = 1
        for n:=num; n>0; n/=10 {
            cnt ^= n&1
        }
    }
    return (num - 1) / 2 + cnt
}
